### ISSUE TITLE
Compatibility with Ember Data Model Fragments new refactor

### ISSUE THREAD

**Issue by patocallaghan** [adopted-ember-addons/ember-data-factory-guy#417]
Ember Data Model Fragments hasn't worked with versions of Ember Data since 3.13+. This is being tracked on their repo in https://github.com/lytics/ember-data-model-fragments/issues/338

To unblock Ember Data upgrades the internals of Model Fragments are being completely refactored to use Ember Data's `RecordData` API. The ongoing work is happening in https://github.com/lytics/ember-data-model-fragments/pull/360/ and they are close to cutting an alpha release.

I tried out the branch in our codebase but we got a load of exceptions from within Model Fragments generated by Factory Guy. It would be great to make sure that Factory Guy is still compatible with this refactored Model Fragments. [I'm going to link a Draft PR]

**Comment by danielspaniel** [comment]
Nice to hear from you Pat. It's been a while. 
I am sorry to say that I will have to leave that update work to someone else ( maybe you folks at intercom can do it ) because I am not using ember or factory guy anymore. I started using clojure ecosystem a few years ago and never looked back.  
 Hopefully you can round up the folks to make this happen.

**Comment by patocallaghan** [comment]
@danielspaniel likewise 😃Its been too long! Yep, I'm happy to dig into it and try fix the issues.

Thinking longer term, considering you're no longer looking to maintain this addon, would you consider putting it for adoption to the [Ember Adopted Addons](https://github.com/adopted-ember-addons) organisation? That's not so pressing an issue I'd be happy to open another issue to discuss that if you think it's worth it.

**Comment by danielspaniel** [comment]
OOoo .. that is good idea Pat, I will look into that adoption thing.

**Comment by patocallaghan** [comment]
I looked into this and I've been able to reproduce an issue where the latest ember-data-factory-guy doesn't work with Ember Data 3.16 and Ember Data Model Fragments 5.0. I'm not entirely sure if the problem is specific to Factory Guy as I've also been able to replicate in our app just clicking but not in this specific repo yet.

I have a test reproduction in this repo here https://github.com/patocallaghan/model-fragments-bug. The specific commit with the changes is in https://github.com/patocallaghan/model-fragments-bug/commit/10dfd71c2244e39a725964e3f85c38af66eebae8

The following test `Unit | Model | grandparent` throws the following error:

```
Error: Assertion Failed: You can only add 'child' fragments or object literals to this property
    at Object.assert (http://localhost:58354/assets/vendor.js:36148:15)
    at http://localhost:58354/assets/vendor.js:98132:120
    at Array.map (<anonymous>)
    at normalizeFragmentArray (http://localhost:58354/assets/vendor.js:98130:17)
    at Class._normalizeData (http://localhost:58354/assets/vendor.js:98192:14)
    at Class.setupData (http://localhost:58354/assets/vendor.js:98426:32)
    at FragmentRecordData.setupFragmentArray (http://localhost:58354/assets/vendor.js:99634:23)
    at FragmentRecordData.getFragmentArray (http://localhost:58354/assets/vendor.js:99608:30)
    at Parent.get (http://localhost:58354/assets/vendor.js:98722:48)
    at http://localhost:58354/assets/vendor.js:15522:32
```

It seems that when we are getting to this [line](https://github.com/richgt/ember-data.model-fragments/blob/f7810c9a2bfbe86aa665b3e488048be65c8101f2/addon/array/fragment.js#L30)
```js
typeOf(data) === 'object' || isInstanceOfType(store.modelFor(type), data)
```
when trying to set up the FragmentArray and the `instance of` check in `normalizeFragmentArray`  the assertion is throwing. This is even though we are trying to add the correct type of fragment to the fragment array.

![image](https://user-images.githubusercontent.com/685034/82708892-4a4fb380-9c77-11ea-936a-f3f309e270d2.png)

**Comment by danielspaniel** [comment]
You think you can fix it @patocallaghan ?

**Comment by patocallaghan** [comment]
@danielspaniel so I've been looking into it are there are some updates on the model fragments PR. This doesn't appear to be a specific issue with Factory Guy as I was able to reproduce the problem directly in application code. The crux of the problem is you can't create fragment array models and then pass them to construct other models. Instead you should use the raw json.

So for example taking a Factory Guy example where `grandparent` is an Ember Data model, `parent` is a Fragment and `children` is a Fragment Array.

This no longer works. It will throw an exception when trying to set the `children` property.
```js
make('grandparent, {
  parent: make('parent, {
    children: makeList('child', 2)
  }
});
```

but this will work
```js
make('grandparent, {
  parent: build('parent, {
    children: buildList('child', 2)
  }
});
```

I'm going to keep this issue open for the time being but will close if its decided that the `make` method should work.

**Comment by patocallaghan** [comment]
@danielspaniel Just an update but I have potential fix for these Model Fragment woes in https://github.com/richgt/ember-data.model-fragments/pull/1

**Comment by danielspaniel** [comment]
looks good Pat, nice work so far

**Comment by patocallaghan** [comment]
Fixed in https://github.com/adopted-ember-addons/ember-data-factory-guy/pull/449
